File: sdk/basin-php/src/Basin/Types/SignUpResult.php
<?php

declare(strict_types=1);

namespace Basin\Types;

/** POST /auth/v1/signup response → {ok, user_id}. */
final class SignUpResult
{
    public function __construct(
        public readonly bool $ok,
        public readonly string $userId,
    ) {
    }

    /** @param array<string,mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            ok: (bool) $data['ok'],
            userId: (string) $data['user_id'],
        );
    }
}

File: sdk/basin-java/src/main/java/io/basin/sdk/SignUpResult.java
package io.basin.sdk;

import com.fasterxml.jackson.annotation.JsonProperty;

/** Response from {@code POST /auth/v1/signup}. */
public final class SignUpResult {

    @JsonProperty("ok")
    public final boolean ok;

    @JsonProperty("user_id")
    public final String userId;

    public SignUpResult(
            @JsonProperty("ok") boolean ok,
            @JsonProperty("user_id") String userId) {
        this.ok = ok;
        this.userId = userId;
    }
}
